### PR TITLE
Feature/dropdown documentation

### PR DESCRIPTION
**Pull request by semia**
- Add a dropdown for Documentation
dropdown

Related to issue #107

### CODE REVIEW

File: app/views/questions/edit.html.haml [hunk 1]
```diff
@@ -27,6 +27,6 @@
     = f.text_field :explanation, class: 'form-control', required: true, id: 'Explication'
   .field
     %h4= f.label 'Documentation'
-    = f.number_field :documentation_id, class: 'form-control', required: true, id: 'Documentation'
+    = f.select(:documentation_id, options_for_select(Documentation.all.map {|d| [d.name, d.id]}), {}, {class: 'form form-group form-control'})
```
Comment (arnlen) [inline app/views/questions/edit.html.haml]: We shouldn't have any DB calls in view.
We can use a Draper Decorator here. ;)

File: app/views/questions/new.html.haml [hunk 2]
```diff
@@ -27,6 +27,6 @@
     = f.text_field :explanation, class: 'form-control', id: 'Explication', required: true
   .field
     %h4= f.label 'Documentation'
-    = f.number_field :documentation_id, class: 'form-control', id: 'Documentation', required: true
+    = f.select(:documentation_id, options_for_select(Documentation.all.map {|d| [d.name, d.id]}), {}, {class: 'form form-group form-control'})
```
Comment (arnlen) [inline app/views/questions/new.html.haml]: Same as above. ;)

File: spec/features/admin/user_edit_spec.rb [hunk 3]
```diff
@@ -1,20 +1,18 @@
...
   let(:user) { create(:user) }
-  let(:admin) { build(:user, :admin) }
+  let(:admin) { create(:user, :admin) }
 
   background do
     sign_in_user(admin)
-    visit "/admin/users/#{user.id}"
-
-    click_on "Modifier l'utilisateur"
+    visit edit_admin_user_path(user.id)
```
Comment (arnlen) [inline spec/features/admin/user_edit_spec.rb]: Nice thx! 👍

File: app/views/questions/edit.html.haml (hunk 1 above)
Comment (semia) [inline app/views/questions/edit.html.haml]: Yes you're right ! :+1: 

I will correct it :-)

File: app/views/questions/edit.html.haml [hunk 4]
```diff
@@ -27,6 +27,6 @@
     = f.text_field :explanation, class: 'form-control', required: true, id: 'Explication'
   .field
     %h4= f.label 'Documentation'
-    = f.number_field :documentation_id, class: 'form-control', required: true, id: 'Documentation'
+    = f.select(:documentation_id, options_for_select(@question.documentation_name(@documentation)), {}, {class: 'form form-group form-control'})
```
Comment (arnlen) [inline app/views/questions/edit.html.haml]: View side:

```
options_for_select(@documentations), {}, {class: 'form form-group form-control'})
```

Controller side:

```
@documentations = Documentation.all.pluck(:name, :id)
```

File: app/decorators/question_decorator.rb [hunk 5]
```diff
@@ -32,4 +32,8 @@ def present_answers(question)
   def answer_correct?(question, index)
     question.correct_answers.include?(question.answers[index])
   end
+
+  def documentation_name(documentation)
```
Comment (arnlen) [inline app/decorators/question_decorator.rb]: To delete. :( snif snif...

File: app/views/questions/new.html.haml [hunk 6]
```diff
@@ -27,6 +27,6 @@
     = f.text_field :explanation, class: 'form-control', id: 'Explication', required: true
   .field
     %h4= f.label 'Documentation'
-    = f.number_field :documentation_id, class: 'form-control', id: 'Documentation', required: true
+    = f.select(:documentation_id, options_for_select(@question.documentation_name(@documentation)), {}, {class: 'form form-group form-control'})
```
Comment (arnlen) [inline app/views/questions/new.html.haml]: Same as above :)

File: spec/decorators/question_decorator_spec.rb [hunk 7]
```diff
@@ -24,7 +26,7 @@
     end
```
Comment (arnlen) [inline spec/decorators/question_decorator_spec.rb]: Shorter version:

```
before { allow(GivenAnswer).to receive(:where).and_return([double, double]) }
it { expect(question.last_answers(number).count).to eq(2) }
```

File: spec/decorators/question_decorator_spec.rb [hunk 8]
```diff
@@ -34,4 +36,27 @@
       it { expect(given_answers).to have_received(:last).with(number) }
     end
   end
+
+  describe '#documentation_name' do
```
Comment (arnlen) [inline spec/decorators/question_decorator_spec.rb]: Good job, but to delete :')

**Comment by arnlen** [comment]
Yeaaah! This is good to merge, as soon as the test are green! 👍

**Pull request merged**